x-pack/plugins/monitoring/server/lib/metrics/classes.js

```diff
@@ -908,15 +903,13 @@ export class BeatsMetric extends Metric {
     super({
       ...opts,
       app: 'beats',
-      uuidField: 'cluster_uuid',
-      timestampField: 'beats_stats.timestamp'
+      ...BeatsMetric.getMetricFields()
     });
   }
 
-  // helper method
   static getMetricFields() {
     return {
-      uuidField: 'cluster_uuid',
+      uuidField: 'beats_stats.beat.uuid',
```

Comment:
I understand why `uuidField` no longer needs to be `cluster_uuid` for Beats metrics. But will it cause any unintended side-effects to change the value of this field to `beats_stats.beat.uuid`? Or is this change actually necessary for things to continue working correctly?

Reply:
Yes, this is necessary now since all the calls to `createBeatsQuery` pass `clusterUuid: clusterUuid` instead of `uuid: clusterUuid` which I think was a mistake in the code earlier, but didn't actually cause an issue. At least not one that I can detect, unless we can see from the functional test results (tests are still running as I am writing this).

In the bigger picture though, I think it's starting to be a bad thing that we have `getMetricsFields` and a generic `createQuery`. I'd like to have `createXyzQuery` functions for all the products and have them be more in charge of how queries work. We kind of started this with `createBeatsQuery` but it still depends on `createQuery`, which might also be a bad thing.